scripts/add_directory.js

```diff
@@ -0,0 +1,287 @@
+    }
+    if (fadeTimeout) {
+        clearTimeout(fadeTimeout);
+        resText.css("display: none;");
+    }
+    resText[0].className = textClass;
+    resText.text(text);
+    resText.fadeIn("fast");
+    fadeTimeout = setTimeout(function() {
+        resText.fadeOut();
+        fadeTimeout = null;
+    }, 5000);
+}
+
+function createBatchUi(folder, num) {
+    return `<div data-batch="${num}" style="display: none;" class="batch panel panel-primary">
+            <div class="panel-heading">
+                <h3 id="batch-dir-${num}" class="panel-title">${folder}</h3>
+            </div>
+            <div class="panel-body">
+                <div class="row">
+                    <span class="col-xs-5">
+                        <div class="row">
+                            <span class="col-sm-4 col-xs-12">
+                                <label for="name-input-${num}">Batch Name</label>
+                            </span>
+                            <span class="col-sm-8 col-xs-12">
+                                <input id="name-input-${num}" type="text" placeholder="Batch name...">
+                            </span>
+                        </div>
+                    </span>
+                    <span class="col-xs-4">
+                        <input id="recursive-check-${num}" type="checkbox">
+                        <label for="recursive-check-${num}">Add recursively</label>
+                    </span>
+                    <span class="col-xs-3">
+                        <div class="row">
+                            <span class="col-sm-6 col-xs-12">
+                                <button id="batch-submit-${num}" class="btn btn-success" onclick="submitBatch(${num})"><span class="hidden rotating glyphicon glyphicon-repeat"></span>Create</button>
+                            </span>
+                            <span class="col-sm-6 col-xs-12">
+                                <button id="batch-cancel-${num}" class="btn btn-danger" onclick="deleteBatch(${num})">Cancel</button>
+                            </span>
+                        </div>
+                    </span>
+                </div>
+                <div class="alert-wrapper" id="alert-wrapper-${num}" class="row"></div>
+            </div>
+        </div>`;
+}
+
+function submitBatch(n) {
+    let batchReq = {
+        dir: $("#name-input-" + n).val(),
+        recursive: $("#recursive-check-" + n)[0].checked,
+        batch_name: $("#batch-dir-" + n).text()
+    };
+    let buttons = $(`[data-batch=${n}] button`);
+    buttons.toggleClass("disabled");
+    $(".glyphicon", buttons).toggleClass("hidden");
+    // eventually this will make an actual post request
+    setTimeout(function() {
+        res = [{result: "success", err_msg: null, img_errs: []},
+            {result: "error", err_msg: "Unhashable images detected", img_errs: ["02_98.png", "03_98.png"]},
+            {result: "fail", err_msg: "Database unreachable", img_errs: []}

```

Comment:
each response will be mapped to the right function call that sent the request automatically, so the response doesn't need to include the batch id. I'll change the front end code to support different errors for each image, keep that part of the backend the same